packages/@idemsInternational/env-replace/src/index.ts

```diff
@@ -0,0 +1,182 @@
+  rawGlobInput: string;
+
+  /** Additional options to provide directly to input glob match */
+  rawGlobOptions: GlobOptionsWithFileTypesUnset;
+
+  /** Throw an error if environment variable is missing. Default true */
+  throwOnMissing: boolean;
+}
+
+const DEFAULT_CONFIG: IEnvReplaceConfig = {
+  cwd: process.cwd(),
+  envAdditional: {},
+  excludeFolders: ["node_modules/**"],
+  excludeVariables: [],
+  fileNameFind: ".template.",
+  fileNameReplace: ".",
+  includeFolders: ["**"],
+  includeVariables: [],
+  rawGlobInput: "",
+  rawGlobOptions: {},
+  throwOnMissing: true,
+};
+
+/**
+ * Regex pattern used to identify templated variable names.
+ * Supports minimal alphanumeric and underscore characters.
+ * Uses capture group to detect variable name within ${VAR_NAME} syntax
+ */
+const VARIABLE_NAME_REGEX = /\${([a-z0-9_]*)}/gi;
+
+/**
+ * Utility class to handle replacing placeholder templates with environment variables
+ * Inspired by https://github.com/danday74/envsub
+ *
+ * @example
+ * ```ts
+ * import { envReplace, IEnvReplaceConfig } from "@idemsInternational/env-replace";
+ *
+ * // default config processes `.template.` files using process env
+ * const config: IEnvReplaceConfig = {}
+ * envReplace.replaceFiles(config)
+ * ```
+ * @see IEnvReplaceConfig for full configuration options
+ *
+ */
+class EnvReplaceClass {
+  private config: IEnvReplaceConfig;
+  private globalEnv: Record<string, any>;
+  private summary: { [inputName: string]: { [variableName: string]: any } };
+
+  /**
+   * Replace templated variables in files, as matched via config
+   * @returns list of variables replaced, with full output written to file
+   */
+  public async replaceFiles(config: Partial<IEnvReplaceConfig>) {
+    this.config = { ...DEFAULT_CONFIG, ...config };
+    this.globalEnv = { ...process.env, ...this.config.envAdditional };
+    this.summary = {};
+
+    const { excludeFolders, cwd, rawGlobOptions, fileNameFind, fileNameReplace } = this.config;
+
+    const inputGlob = this.generateInputGlob();
+    const inputNames = await glob(inputGlob, {
+      ignore: excludeFolders,
+      cwd,
+      dot: true,
+      posix: true,
+      ...rawGlobOptions,
+    });
+
+    for (const inputName of inputNames) {
+      const filepath = resolve(cwd, inputName);
+      const sourceContent = readFileSync(filepath, { encoding: "utf-8" });
+
+      // determine variables to replace and values to replace with
+      const variablesToReplace = this.generateReplacementList(sourceContent);
+      const replacementEnv = this.generateReplacementEnv(variablesToReplace);
+
+      // handle replacement
+      const outputName = inputName.replace(fileNameFind, fileNameReplace);
+      this.summary[outputName] = {};
+
+      const replaceContent = this.generateReplaceContent(outputName, sourceContent, replacementEnv);
+      const outputPath = resolve(cwd, outputName);
+      writeFileSync(outputPath, replaceContent);
+    }
+    return this.summary;
```

Comment:
It could be nice to log the paths of the output files here (the keys of `this.summary`) to see what's been generated? Although these files won't necessarily have changed, so perhaps git diffs are a better place to view changes

Reply:
I had previously included a table log, although was concerned whether github actions would still be able to correctly mask any variables populated from secrets if logging after the various manipulations. No harm including as an optional config param to test later however, I'll include the update with next pr (screenshot below for reference)

![image](https://github.com/IDEMSInternational/parenting-app-ui/assets/10515065/8b79ec51-951d-476b-8561-80c076df245e)